File: Eazy_Store/src/main/java/com/eazybytes/eazystore/service/CategoryService.java
package com.eazybytes.eazystore.service;

import java.util.List;

import com.eazybytes.eazystore.dto.CategoryRequest;
import com.eazybytes.eazystore.dto.CategoryResponse;

public interface CategoryService {

    CategoryResponse createCategory(CategoryRequest request);
    List<CategoryResponse> getAllCategories();
    CategoryResponse updateCategory(Long id, CategoryRequest request);
    void deleteCategory(Long id);
}

File: frontend/src/services/categoryService.js
import {
  getAllCategories,
  getCategoryById,
  createCategory,
  updateCategory,
  deleteCategory,
} from "../api/categoryApi";

const categoryService = {
  getAllCategories,
  getCategoryById,
  createCategory,
  updateCategory,
  deleteCategory,
};

export default categoryService;
